packages/block-editor/src/components/block-tools/block-selection-button.js

```diff
@@ -71,17 +74,33 @@ function BlockSelectionButton( { clientId, rootClientId }, ref ) {
 				getNextBlockClientId,
 				getPreviousBlockClientId,
 				canMoveBlock,
+				getBlockParents,
 			} = unlock( select( blockEditorStore ) );
 			const { getActiveBlockVariation, getBlockType } =
 				select( blocksStore );
 			const index = getBlockIndex( clientId );
-			const { name, attributes } = getBlock( clientId );
+			const block = getBlock( clientId );
+			const { name, attributes } = block;
 			const blockType = getBlockType( name );
 			const orientation =
 				getBlockListSettings( rootClientId )?.orientation;
 			const match = getActiveBlockVariation( name, attributes );
 
-			const blockBindings = attributes?.metadata?.bindings;
+			const blockBindings = replacePatternOverrideDefaultBindings(
+				name,
+				attributes?.metadata?.bindings
+			);
+
+			if ( blockBindings?.content?.source === 'core/pattern-overrides' ) {
+				const parents = getBlockParents( clientId );
+				const parentBlock =
+					parents && parents.length > 0
+						? getBlock( parents[ 0 ] )
+						: null;
+				blockContext[ 'pattern/overrides' ] =
+					parentBlock?.attributes?.content;
+			}
```

Comment:
This doesn't seem like the most elegant solution to me. Is there another way to get the pattern attributes other than to look up the values manually and insert them like this?

Also, I'm unsure if we can assume that the first parent will be the pattern. Gutenberg seems to flatten the hierarchy of blocks in pattern instances, so this appears to work, but it'd be great to additional insight on this 🙏

Reply:
> Also, I'm unsure if we can assume that the first parent will be the pattern. Gutenberg seems to flatten the hierarchy of blocks in pattern instances, so this appears to work, but it'd be great to additional insight on this

Yep I think you'll need to iterate through all block parents to find the `core/block` block. Also need to take into account a pattern within a nested pattern, and ensure you find the nearest one.

There's a few places with this kind of code 😅 

Just to note, this component would be removed if #65204 lands as is.